### PR TITLE
Initialize CHANGELOG section

### LINKED ISSUE #131: Initialize CHANGELOG
We are anticipating updating the sample metadata that is available on the portal (in addition to updating the contents of the SCE objects that are available on the portal). As we anticipate future changes to things on the portal, it might be helpful to include a change log. 

From Deepa: 
> If we are expecting similar metadata changes to happen again in the future,  I was thinking something like release notes, but change log is better since we aren’t really sharing versions explicitly on the portal. In addition to including it downloads, we should have it in the docs with the date.

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
Closes #131. Here, I'm adding the initial version of the CHANGELOG. Per comments on that issue, this PR:

* Initializes the CHANGELOG file
* Adds intro text that covers where to find more information about downloads (i.e., workflow version and download generation date)
* Puts the CHANGELOG right after the FAQ section

~~Draft for now until I check the build and spell check. After that, I'll mark it as ready for review.~~ Ready for review!

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -11,5 +11,6 @@ download_files
 sce_file_contents
 getting_started
 faq
+CHANGELOG
```
Comment (dvenprasad) [inline docs/index.md:14]: Oh I forgot we had citation in there already! I was looking at production. 
```suggestion
citation
```

File: docs/index.md [hunk 2]
```diff
@@ -11,5 +11,6 @@ download_files
...
 citation
```
Comment (dvenprasad) [inline docs/index.md:15]:
```suggestion
CHANGELOG
```
Moving citation above CHANGELOG

**Review by dvenprasad — APPROVED** [review]
I switched the positions of citations and CHANGELOG, so CHANGELOG is the last item on the navbar. 

I forgot we'd added citations to the docs.

**Comment by jaclyn-taroni** [comment]
I'm going to get this merged in case we want to do a release tomorrow while I'm OOO.

**Pull request merged**